pkg/subctl/cmd/validate_deployment.go

```diff
@@ -17,46 +17,128 @@ package cmd
+	exitOnError("Unable to get the Submariner client", err)
+
+	localClusterName := submariner.Status.ClusterID
+	endpointList, err := submarinerClient.SubmarinerV1().Endpoints(submariner.Namespace).List(metav1.ListOptions{})
+	if err != nil {
+		exitOnError("Error while listing the endpoints", err)
 	}
 
-	kubeClientSet, err := kubernetes.NewForConfig(config)
+	status.Start("* Verifying if cluster CIDRs overlap")
+
+	var localEndpoint submarinerv1.Endpoint
+	overlappingClusters := make(map[string]submarinerv1.Endpoint)
+	for i, source := range endpointList.Items {
+		if localClusterName == source.Spec.ClusterID {
+			localEndpoint = source
+		}
+
+		for _, dest := range endpointList.Items[i+1:] {
+			// Currently we dont support multiple endpoints in a cluster, hence return an error.
+			// When the corresponding support is added, this check needs to be updated.
+			if source.Spec.ClusterID == dest.Spec.ClusterID {
+				message = fmt.Sprintf("Looks like some stale endpoints are present for cluster %q",
+					source.Spec.ClusterID)
+				status.QueueFailureMessage(message)
+				status.End(cli.Failure)
+				break
+			}
+
+			for _, subnet := range dest.Spec.Subnets {
+				overlap, err := util.IsOverlappingCIDR(source.Spec.Subnets, subnet)
+				if err != nil {
+					// Ideally this case will never hit, as the subnets are valid CIDRs
+					message = fmt.Sprintf("Error parsing the CIDR %s in cluster %q", err, dest.Spec.ClusterID)
+					status.QueueFailureMessage(message)
+					status.End(cli.Failure)
+					break
+				}
+
+				if overlap {
+					overlappingClusters[source.Spec.ClusterID] = source
+					overlappingClusters[dest.Spec.ClusterID] = dest
+				}
+			}
+		}
+	}
+
+	if len(overlappingClusters) > 0 {
+		if _, exists := overlappingClusters[localClusterName]; exists {
+			delete(overlappingClusters, localClusterName)
+			message = fmt.Sprintf("localCluster %q with CIDRs %q overlaps with clusters %#v",
+				localClusterName, localEndpoint.Spec.Subnets, overlappingClusters)
+		} else {
+			message = fmt.Sprintf("localCluster %q does not have overlapping CIDRs with other member clusters. "+
+				"However, the following submariner member clusters have overlapping CIDRs %#v", localClusterName,
+				getClusterDetails(overlappingClusters))
+		}
+		status.QueueFailureMessage(message)
+		status.End(cli.Failure)
+	}
+
+	status.QueueSuccessMessage("Clusters do not have overlapping CIDRs")
+	status.End(cli.Success)
+}
+
+func getClusterDetails(overlappingClusters map[string]submarinerv1.Endpoint) string {
+	output := []string{}
+	for _, cluster := range overlappingClusters {
+		output = append(output, fmt.Sprintf("clusterID: %s, subnets: %s", cluster.Spec.ClusterID, cluster.Spec.Subnets))
+	}
+	return strings.Join(output, " ; ")
+}
+
+func checkPods(item restConfig, submariner *v1alpha1.Submariner, operatorNamespace string) {
+	message := fmt.Sprintf("* Validating submariner pods in %q", item.clusterName)
```

Comment:
In other validation commands the message is without `*` prefix

Reply:
Yes, I know ;-)

I added it looking at `subctl join ...` where beginning of some operation (aka validation) is shown with some dot 

```
2021-03-22T15:35:52.6449258Z [cluster1]  • Discovering network details  ...
2021-03-22T15:35:52.6449880Z [cluster1]     Discovered network details:
2021-03-22T15:35:52.6450610Z [cluster1]         Network plugin:  generic
2021-03-22T15:35:52.6451177Z [cluster1]         Service CIDRs:   [100.0.0.0/16]
2021-03-22T15:35:52.6451675Z [cluster1]         Cluster CIDRs:   [10.0.0.0/16]
2021-03-22T15:35:52.6452442Z [cluster1]  ✓ Discovering network details
2021-03-22T15:35:52.6453226Z [cluster1]  • Discovering multi cluster details  ...
2021-03-22T15:35:52.6454078Z [cluster1]  • Validating Globalnet configurations  ...
2021-03-22T15:35:52.6454960Z [cluster1]  ✓ Validating Globalnet configurations
2021-03-22T15:35:52.6455736Z [cluster1]  • Assigning Globalnet IPs  ...
2021-03-22T15:35:52.6456461Z [cluster1]  ✓ Assigning Globalnet IPs
2021-03-22T15:35:52.6457157Z [cluster1]  ✓ GlobalCIDR is: 169.254.1.0/24
2021-03-22T15:35:52.6457903Z [cluster1]  ✓ Discovering multi cluster details
2021-03-22T15:35:52.6458709Z [cluster1]  • Deploying the Submariner operator  ...
2021-03-22T15:35:52.6459521Z [cluster1]  ✓ Deploying the Submariner operator
2021-03-22T15:35:52.6460392Z [cluster1]  ✓ Created Lighthouse service accounts and roles
2021-03-22T15:35:52.6461487Z [cluster1]  • Creating SA for cluster  ...
2021-03-22T15:35:52.6464075Z [cluster1]  ✓ Creating SA for cluster
2021-03-22T15:35:52.6464793Z [cluster1]  • Deploying Submariner  ...
2021-03-22T15:35:52.6465511Z [cluster1]  ✓ Deploying Submariner
2021-03-22T15:35:52.6466220Z [cluster1]  ✓ Submariner is up and running
```

If people like it, my idea was to change all the other commands. But anyway, I can modify if required, no issues.